src/common/src/types/interval.rs

```diff
@@ -81,23 +81,40 @@ impl IntervalUnit {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
-    /// Justify interval, convert 1 month to 30 days and 86400 s to 1 day.
-    /// If day is positive, complement the seconds negative value.
-    /// These rules only use in interval comparison.
-    pub fn justify_interval(&mut self) {
-        #[expect(deprecated)]
-        let total_usecs = self.as_usecs_i64();
-        *self = Self {
-            months: 0,
-            days: (total_usecs / USECS_PER_DAY) as i32,
-            usecs: total_usecs % USECS_PER_DAY,
+    /// Justify interval, convert 24 hours to 1 day and 30 days to 1 month.
+    /// Also makes signs of all fields to be the same.
+    ///
+    /// <https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/timestamp.c#L2740>
+    pub fn justify_interval(&self) -> Option<Self> {
```

Comment:
This is currently not exposed by frontend as `justify_interval` yet. We just added unit tests to make sure it returns the same result as `IntervalCmpValue::as_justified`.

Reply:
Actually they do not always return the same results ... This `justify_interval` from PostgreSQL has a false positive error that `as_justified` can handle correctly.

Reply:
If interval has 30 days, after it is justified it will return 1 month.
In arithmetic with timestamp, will we convert it back to 30 days for correctness?

Reply:
Unfortunately, no. We would not be able to convert back because it is not injective. The original value might be '-45 days' or '-1 months -15 days' or even '1 months -75 days' but they all return the same '-45 days'.

FYI This wired behavior of equality is not limited to arithmetic or interval in PostgreSQL, where x == y may lead to f(x) <> f(y). https://github.com/risingwavelabs/risingwave/issues/3863#issuecomment-1184067727